cpp/fetcher/remote_peer.cc

```diff
@@ -73,12 +100,12 @@ RemotePeer::RemotePeer(
     unique_ptr<AsyncLogClient>&& client, unique_ptr<LogVerifier>&& verifier,
     const std::function<void(const ct::SignedTreeHead&)>& on_new_sth,
     util::Task* task)
-    : Peer(move(client)), impl_(new Impl(move(verifier), on_new_sth, task)) {
+    : Peer(move(client)),
+      impl_(new Impl(move(verifier), client_.get(), on_new_sth, task)) {
   TaskHold hold(task);
   task->DeleteWhenDone(impl_);
 
-  shared_ptr<SignedTreeHead> new_sth(make_shared<SignedTreeHead>());
-  client_->GetSTH(new_sth.get(), bind(&Impl::DoneGetSTH, impl_, new_sth, _1));
+  impl_->FetchSTH(task);
```

Comment:
This also looked weird to me, as it feels like `FetchSTH` would be liable to either try to get a status out of `task` (which wouldn't work, as it's not a done task), or "take it" to do the work and eventually call `Return()` on it (which would cause everything to fall apart!)...

If you apply my previous suggestion of just using `task_` instead of `task` in `FetchSTH`, you could just pass in `nullptr` here without problem.